### PR TITLE
Fixes + diesel features

### PR DESCRIPTION
**Pull request by Xoov**
Minor bug fix, features for diesel engine

### CODE REVIEW

File: parser.cpp [hunk 1]
```diff
@@ -245,7 +245,7 @@ std::string cParser::readToken( bool ToLower, const char *Break ) {
     // launch child parser if include directive found.
     // NOTE: parameter collecting uses default set of token separators.
 	if( expandIncludes && token == "include" ) {
-        std::string includefile = readToken(ToLower); // nazwa pliku
+		std::string includefile = deserialize_random_set(*this); //readToken(ToLower); // nazwa pliku
```
Comment (Milek7) [inline parser.cpp]: Don't do this, non-deterministic parsing is unacceptable. Handle it in the consumer.

**Review by Milek7 — CHANGES_REQUESTED** [review]

File: parser.cpp (hunk 1 above)
Comment (Xoov) [reply parser.cpp]: I don't get it - you don't like the feature (including randomly one of several files) or the way it is implemented?

Comment (Xoov) [reply parser.cpp]: Random includes are now limited only to mmd file, which has nothing to do with core simulation nor scenery.

**Comment by Milek7** [comment]
I still think random includes are horrible, but let's accept this for now as this feature is already used on repository and replace it later.

**Pull request merged**